doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,317 @@
+.. _soc_porting_guide:
+
+SoC Porting Guide
+###################
+
+To add Zephyr support for a new :term:`SoC`, you at least need a *SoC directory*
+with various files in it. Also a SoC :file:`.dtsi` in
+:zephyr_file:`dts/<ARCH>/<VENDOR>` is required.
+
+SoC Definitions
+***************
+
+It is expected that you are familiar with the board concept in Zephyr.
+A high level overview of the hardware support hierarchy and terms used in the
+Zephyr documentation can be seen in :ref:`hw_support_hierarchy`.
+
+For SoC porting, the most important terms are:
+
+- SoC: the exact system on a chip the board's CPU is part of
+- SoC series: a smaller group of tightly related SoCs
+- SoC family: a wider group of SoCs with similar characteristics
+- CPU Cluster: a cluster of one or more CPU cores.
+- CPU core: a particular CPU in an architecture
+- Architecture: an instruction set architecture
```

Comment:
This is great but maybe we could use a more logical order:
- Architecture: an instruction set architecture
- SoC family: a group of SoCs with similar characteristics (typically provided by a single vendor)
- SoC series: a smaller group of tightly related SoCs
- SoC: the exact system on a chip the board's CPU is part of
- CPU core: a particular CPU in an architecture
- CPU Cluster: a cluster of one or more CPU cores.

Reply:
exactly why is that order more logical ?

I guess there are good arguments for different orders, I just would like to know why you think that order is better in this case.

So that others may give opinions and to avoid changing back and forth if other would like a different order.

Reply:
Actually, what caught my attention are rather the 2 first, as in the second you mention smaller while first is only one soc.
At least removing `smaller` would be required.
After that, I agree that everything is subjective.

Reply:
>  At least removing smaller would be required.

Done